frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -176,6 +187,19 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   const [error, setError] = useState('');
   const [mousePosition, setMousePosition] = useState<Coordinate>({ x: 0, y: 0 });
   const [hoverVisible, setHoverVisible] = useState<boolean>(false);
+  const [tagModal, setTagModal] = useState<boolean>(false);
+  const [tagName, setTagName] = useState('');
+  const [allTags, setAllTags] = useState<ProblemTag[]>([]);
+
+  useEffect(() => {
+    getAllProblemTags()
+      .then((res) => {
+        setAllTags(res);
+      })
+      .catch((err) => {
+        setError(err.message);
+      });
+  }, []);
```

Comment:
Is it possible to include the majority of this tag logic into a separate component within the same problem folder? Main reason being that this file is already really large and I'm thinking in a future PR where we improve the creation flow, it might break it into multiple steps/pages

Reply:
Done! ✅ 

I think separating the creation flow is an interesting idea for sure, let me know what thoughts you have! At least separating the components in the `ProblemDisplay` folder would be good too. 👍